OpenAI agents tests (#898)
temporalio/sdk-python: temporalio/contrib/openai_agents/README.md

```mermaid
graph TD
    A["INPUT"] --> B["LLM"]
    B <--> C["TOOLS"]
    B --> D["OUTPUT"]
```

To provide durable execution, Temporal needs to be able to recover from failures at any step of this process.
To do this, Temporal requires separating an application's deterministic (repeatable) and non-deterministic parts:

1. Deterministic pieces, termed *workflows*, execute the same way if re-run with the same inputs.
2. Non-deterministic pieces, termed *activies*, have no limitations—they may perform I/O and any other operations.

Temporal maintains a server-side execution history of all state state passing in and out of a workflow, using it to recover when needed. 
See the [Temporal documentation](https://docs.temporal.io/evaluate/understanding-temporal#temporal-application-the-building-blocks) for more information.

How do we apply the Temporal execution model to enable durable execution for AI agents?

- The core control flow, which is managed by the OpenAI Agents SDK, goes into a Temporal workflow.
- Calls the the LLM provider, which are inherently non-deterministic, go into activities.
- Calls to tools, which could contain arbitrary code, similarly go into activities.

> This module ensures that LLM calls and tool calls originating from the OpenAI Agents SDK in run as Temporal activities.

Reviewer: ```suggestion
This module ensures that LLM calls and tool calls originating from the OpenAI Agents SDK run as Temporal activities.
```

@workflow.defn
class HelloWorldAgent:
    @workflow.run
    async def run(self, prompt: str) -> str:
        agent = Agent(
            name="Assistant",
            instructions="You only respond in haikus.",
        )

        result = await Runner.run(agent, input=prompt)
        return result.final_output
```

If you are familiar with Temporal and with Open AI Agents SDK, this code will look very familiar.
We annotate the `HelloWorldAgent` class with `@workflow.defn` to define a workflow, then use the `@workflow.run` annotation to define the entrypoint.
We use the `Agent` class to define a simple agent, one always responds with haikus.
Within the workflow, we start agent using the `Runner`, as is typical, passing through `prompt` as an argument.

Perhaps the most interesting thing about this code is the `with workflow.unsafe.imports_passed_through():` statement that precedes the OpenAI Agents SDK imports.
This statement tells Temporal to skip sandboxing for these trusted libraries.
This is important because Python's dynamic nature forces Temporal's Python's sandbox to re-validates imports every time a workflow runs, which comes at a performance cost.

> The OpenAI Agents SDK also contains certain code that Temporal is not able to validate automatically for determinism.

Author: @jssmith is still working on the readme as well.

from temporalio.worker import Worker

from hello_world_workflow import HelloWorldAgent

async def worker_main():
    # Configure the OpenAI Agents SDK to use Temporal activities for LLM API calls
    # and for tool calls.
    with set_open_ai_agent_temporal_overrides():
        # Create a Temporal client connected to server at the given address
        # Use the OpenAI data converter to ensure proper serialization/deserialization
        client = await Client.connect(
            "localhost:7233",
            data_converter=open_ai_data_converter,
        )

        model_activity = ModelActivity(model_provider=None)
        with concurrent.futures.ThreadPoolExecutor(max_workers=10) as activity_executor:
            worker = Worker(
                client,
                task_queue="my-task-queue",
                workflows=[HelloWorldAgent],
                activities=[model_activity.invoke_model_activity],
                activity_executor=activity_executor,
            )
            await worker.run()

if __name__ == "__main__":
    asyncio.run(worker_main())
```

We wrap the entire `worker_main` function body in the `set_open_ai_agent_temporal_overrides()` context manager.

> This creates a Temporal activity whenever the OpenAI Agents SDK invokes an LLM or calls a tool.

Reviewer: ```suggestion
This causes a Temporal activity to be invoked whenever the OpenAI Agents SDK invokes an LLM or calls a tool.
```

?

We also pass the `open_ai_data_converter` to the Temporal Client, which ensures proper serialization of OpenAI Agents SDK data.
We also create a `ModelActivity` which serves as a generic wrapper for LLM calls, and we register this wrapper's invocation point, `model_activity.invoke_model_activity`, with the workflow.

> In order to launch the agent, use the standard Temporal workflow invocation.

Reviewer: ```suggestion
In order to launch the agent, use the standard Temporal workflow invocation:
```
